JSCore06/reverse.js

```
@@ -0,0 +1,46 @@
+function mergeOptions(firstToMerge, secondToMerge){
+    var mergedObj = {};
+    for (var attrname in firstToMerge) mergedObj[attrname] = firstToMerge[attrname];
+    for (var attrname in secondToMerge) mergedObj[attrname] = secondToMerge[attrname];
+    return mergedObj;
+}
+
+function reverseCalc(string, extraOperators) {
+  var stack = [];
+  var character;
+  var baseOperators = {
+    "+": function (a, b) { return a + b },
+    "-": function (a, b) { return a - b },
+    "*": function (a, b) { return a * b }
+  };
+
+  var operators = extraOperators
+  ? mergeOptions(baseOperators, extraOperators)
+  : baseOperators;
+
+  var isProperString = /^[0-9*+ ().,-]+$/.test(string);
+  if (!string || !isProperString ) {
+    throw new Error("Please put an equation to solve in a form of string with spaces");
+  } else {
+    var arr = string.split(" ");
+    for (var k = 0; k < string.length;  k++) {
+      character = arr[k];
+
+      if (/^\d+$/.test(character)){
+        stack.push(character);
+      } else if (character in operators) {
+        var secondOperand = parseInt(stack.pop());
+        var firstOperand = parseInt(stack.pop());
+        var value = operators[character](firstOperand, secondOperand);
+        stack.push(value);
+      }
```

We also need to add `else` statement which should throw an error because if it is not a number and an operator it is some incorrect character. For example:
```
node runner.js "( 2 2 +"
[ 4 ]
Your ES6 result is: 4
Your ES5 result is: 4
```
It works but it shouldn't.